File: GameLibrary/src/cc/lib/checkerboard/AIStats.java
package cc.lib.checkerboard;

import cc.lib.game.Utils;

public class AIStats {

    final long startTime;
    long prunes=0;
    long evalCount = 0;
    long evalTimeTotalMSecs = 0;
    long [] pieceTypeCount = new long[PieceType.values().length];
    double [] pieceTypeValue = new double[PieceType.values().length];

    AIStats() {
        startTime = System.currentTimeMillis();
    }

    @Override
    public String toString() {

        String s = "Run Tiime: " + Utils.formatTime(System.currentTimeMillis()-startTime)
                + "\nprunes=" + prunes
                + "\nPruned %" + (100f * prunes / evalCount)
                + "\nevalCount=" + evalCount
                + "\nevalTimeTotalMSecs=" + evalTimeTotalMSecs;

        for (int i=0; i<pieceTypeValue.length; i++) {
            if (pieceTypeCount[i] > 0)
                s += String.format("\n%-20s AVG: %5.3f", PieceType.values()[i], (pieceTypeValue[i] / pieceTypeCount[i]));
        }


        return s;
    }
}

File: Checkerboard/src/cc/lib/checkerboard/AIStats.kt
package cc.lib.checkerboard

import cc.lib.game.Utils

class AIStats internal constructor() {
	val startTime: Long
    var prunes: Long = 0
    var evalCount: Long = 0
    var evalTimeTotalMSecs: Long = 0
    var pieceTypeCount = LongArray(PieceType.values().size) { 0 }
    var pieceTypeValue = DoubleArray(PieceType.values().size) { 0.0 }
	override fun toString(): String {
		var s = """
	    	Run Time: ${Utils.formatTime(System.currentTimeMillis() - startTime)}
	    	prunes=$prunes
	    	Pruned %${100f * prunes / evalCount}
	    	evalCount=$evalCount
	    	evalTimeTotalMSecs=$evalTimeTotalMSecs
	    	""".trimIndent()
		for (i in pieceTypeValue.indices) {
			if (pieceTypeCount[i] > 0) s += String.format("\n%-20s AVG: %5.3f", PieceType.values()[i], pieceTypeValue[i] / pieceTypeCount[i])
		}
		return s
	}

	init {
		startTime = System.currentTimeMillis()
	}
}